pkg/agent/cluster.go

```diff
@@ -516,3 +521,49 @@ func (czero *Cluster) humanFriendlyClusterInstallStatus(status string) string {
 		return clusterStoppedInstallingStates[status]
 	}
 }
+
+// AddRouterCAToClusterCA adds router CA to cluster CA in kubeconfig.
+func AddRouterCAToClusterCA(ctx context.Context, config *rest.Config, directory string) (err error) {
```

Comment:
In any case it looks that the `agent wait for` command does not honour this step like the regular `wait for` (and I'm not sure why it wasn't added originally), so I think that the proper fix for that would be to add the already existing [addRouterCAToClusterCA](https://github.com/openshift/installer/blob/d96ffaeb9cb7b31c0a8e4516080b0fb6fd2d332c/cmd/openshift-install/create.go#L347) (by refactoring it so that it could be accessible from both the command, to avoid the code duplication) and reusing it [here](https://github.com/openshift/installer/blob/d96ffaeb9cb7b31c0a8e4516080b0fb6fd2d332c/pkg/agent/cluster.go#L327)

Reply:
@andfasano As you suggested, I have refactored the code. I have added the function under pkg/utils/utils.go to be accessible for both the commands. Please let me know where to add this function if it isn't supposed to be in pkg/utils/utils.go